ethereum/test/foundry/unit/concrete/Executor/_Executor_Shared.t.sol

```diff
@@ -0,0 +1,220 @@
+import {DiamondInit} from "../../../../../cache/solpp-generated-contracts/zksync/DiamondInit.sol";
+import {DiamondProxy} from "../../../../../cache/solpp-generated-contracts/zksync/DiamondProxy.sol";
+import {VerifierParams} from "../../../../../cache/solpp-generated-contracts/zksync/Storage.sol";
+import {ExecutorFacet} from "../../../../../cache/solpp-generated-contracts/zksync/facets/Executor.sol";
+import {GettersFacet} from "../../../../../cache/solpp-generated-contracts/zksync/facets/Getters.sol";
+import {GovernanceFacet} from "../../../../../cache/solpp-generated-contracts/zksync/facets/Governance.sol";
+import {MailboxFacet} from "../../../../../cache/solpp-generated-contracts/zksync/facets/Mailbox.sol";
+import {IExecutor} from "../../../../../cache/solpp-generated-contracts/zksync/interfaces/IExecutor.sol";
+import {Diamond} from "../../../../../cache/solpp-generated-contracts/zksync/libraries/Diamond.sol";
+
+contract ExecutorTest is Test {
+    address internal constant L2_SYSTEM_CONTEXT_ADDRESS = 0x000000000000000000000000000000000000800B;
+    address internal constant L2_KNOWN_CODE_STORAGE_ADDRESS = 0x0000000000000000000000000000000000008004;
+    address internal constant L2_TO_L1_MESSENGER = 0x0000000000000000000000000000000000008008;
+
+    address internal owner;
+    address internal validator;
+    address internal randomSigner;
+    AllowList internal allowList;
+    GovernanceFacet internal governance;
+    ExecutorFacet internal executor;
+    GettersFacet internal getters;
+    MailboxFacet internal mailbox;
+    bytes32 internal newCommittedBlockBlockHash;
+    bytes32 internal newCommittedBlockCommitment;
+    uint256 internal currentTimestamp;
+    IExecutor.CommitBlockInfo internal newCommitBlockInfo;
+    IExecutor.StoredBlockInfo internal newStoredBlockInfo;
+
+    IExecutor.StoredBlockInfo internal genesisStoredBlockInfo;
+    IExecutor.ProofInput internal proofInput;
+
+    function getGovernanceSelectors() private view returns (bytes4[] memory) {
+        bytes4[] memory selectors = new bytes4[](5);
+        selectors[0] = governance.setPendingGovernor.selector;
+        selectors[1] = governance.acceptGovernor.selector;
+        selectors[2] = governance.setValidator.selector;
+        selectors[3] = governance.setPorterAvailability.selector;
+        selectors[4] = governance.setPriorityTxMaxGasLimit.selector;
+        return selectors;
+    }
+
+    function getExecutorSelectors() private view returns (bytes4[] memory) {
+        bytes4[] memory selectors = new bytes4[](4);
+        selectors[0] = executor.commitBlocks.selector;
+        selectors[1] = executor.proveBlocks.selector;
+        selectors[2] = executor.executeBlocks.selector;
+        selectors[3] = executor.revertBlocks.selector;
+        return selectors;
+    }
+
+    function getGettersSelectors() public view returns (bytes4[] memory) {
+        bytes4[] memory selectors = new bytes4[](32);
```

Comment:
These hardcoded selectors are easy to forget to update. Maybe read the ABI from the generated typechain/ABI instead? with `readFile`/`parseJson` it shouldn't be too hard

Reply:
Indeed this is inconvenient, but I'm not aware of any simple solution that can be done using foundry.
For hardhat tests your proposed solution would work, using typescript it can be easily implemented, but unfortunately not with foundry.
But if you have any specific solution in mind please let me know!

Reply:
`readFile`/`parseJson` are available in Foundry, so you can read files directly from Solidity tests